tensornetwork/tn_keras/test_layer.py

```diff
@@ -333,8 +348,45 @@ def test_entangler_num_parameters(dummy_data):
                      input_shape=(data.shape[1],)))
 
   # num_params = entangler_node_params + bias_params
-  expected_num_parameters = num_levels * (leg_dim**4) + leg_dim**num_legs
+  expected_num_parameters = num_levels * (num_legs - 1) * (leg_dim**4) + (
+      leg_dim**num_legs)
+
+  np.testing.assert_equal(expected_num_parameters, model.count_params())
+
+
+@pytest.mark.parametrize('num_levels', list(range(1, 4)))
+@pytest.mark.parametrize('num_legs', list(range(2, 6)))
+@pytest.mark.parametrize('leg_dims', [(4, 8), (8, 4)])
+def test_entangler_asymmetric_num_parameters(num_legs, num_levels, leg_dims):
+  # Disable the redefined-outer-name violation in this function
+  # pylint: disable=redefined-outer-name
+  leg_dim, out_leg_dim = leg_dims
+  data_shape = (leg_dim ** num_legs,)
+
+  # num_levels = 3
+
+  model = Sequential()
+  model.add(
+      DenseEntangler(out_leg_dim**num_legs,
+                     num_legs=num_legs,
+                     num_levels=num_levels,
+                     use_bias=True,
+                     activation='relu',
+                     input_shape=data_shape))
+
+
+  primary = leg_dim
+  secondary = out_leg_dim
+  if leg_dim > out_leg_dim:
+    primary, secondary = secondary, primary
+
+  expected_num_parameters = (num_levels - 1) * (num_legs - 1) * (primary**4) + (
+      num_legs - 2) * primary**3 * secondary + primary**2 * secondary**2 + (
+          out_leg_dim**num_legs)
+
 
+  data = np.random.randint(10, size=(10, data_shape[0]))
+  model(data)
```

Comment:
Can you check that the output shapes are the same as the expected output shapes for all of the permutations of num_levels, num_legs, leg_dims?